pkg/types/baremetal/validation/platform.go

```diff
@@ -413,6 +413,14 @@ func ValidateProvisioning(p *baremetal.Platform, n *types.Networking, fldPath *f
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("clusterProvisioningIP"), p.ClusterProvisioningIP, fmt.Sprintf("%q is not in the provisioning network", p.ClusterProvisioningIP)))
 		}
 
+		// Ensure provisioningNetworkCIDR does not have any host bits set
+		expectedIP := p.ProvisioningNetworkCIDR.IP.Mask(p.ProvisioningNetworkCIDR.Mask)
+		expectedLen, _ := p.ProvisioningNetworkCIDR.Mask.Size()
+		if string(p.ProvisioningNetworkCIDR.IP) != string(expectedIP) {
```

Comment:
I think it may be cleaner to do `if ! p.ProvisioningNetworkCIDR.IP.Equal(expectedIP)` here?

FWIW the reason I initially thought that was I was worried the string comparison wouldn't handle long format ipv6 CIDRs, that's not actually the case but if you wanted to grab the tests to prove it I pushed them to https://github.com/hardys/installer/commit/b487e94a7539e418503cb165e80df32554eb4aed